unstract/task-abstraction/src/unstract/task_abstraction/backends/hatchet.py

```
@@ -0,0 +1,360 @@
+                            )
+
+                        # Create step with parent dependency for sequential execution
+                        if previous_step:
+                            parents = [previous_step]
+                        else:
+                            parents = []
+
+                        @self.hatchet.step(name=step_name, parents=parents)
+                        def workflow_step(context):
+                            # Get the original task function
+                            task_fn = self._tasks[step.task_name]
```

Ruff (0.13.1)

Function definition does not bind loop variable `step`

(B023)